src/features/DebugSession.ts

```diff
@@ -105,7 +114,7 @@ export class PickPSHostProcessFeature implements IFeature {
 
         if (this.waitingForClientToken) {
             this.clearWaitingToken();
-            return this.pickPSHostProcess();
```

Comment:
It will display the UI but the resulting pick is dropped on the floor AFAICT.  As I understand it, the whole command like so:
1. User picks attach to in debug panel
2. Dbg config for attach has `${command.pickPSHostProcess` configured to get the `processId` value
3. That command is invoked and in turn, activates PowerShell since it is registered as activation event
4. When the command completes, VSCode uses the returned `string` as the `processId` value.  If the command does not return a string, the adapter gets sent "undefined" for `processId`.

So what I think we need is a synchronization primitive (like *ResetEvent / CriticalSection) so that the command can "wait /w timeout" on the setLanguageClient to signal that it has been initialized.

BTW in the process of debugging this I found a bug (two actually in VSCode) :-)  https://github.com/Microsoft/vscode/issues/6569#issuecomment-272828999